src/main/java/com/internet/shop/dao/jdbc/OrderDaoJdbcImpl.java

```diff
@@ -0,0 +1,165 @@
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(query);
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                orders.add(getOrderFromResultSet(resultSet));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Failed to get orders from database ", e);
+        }
+        for (Order order : orders) {
+            order.setProducts(getProductsFromOrder(order.getId()));
+        }
+        return orders;
+    }
+
+    @Override
+    public Order update(Order order) {
+        String query = "DELETE FROM order_products WHERE order_id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(query);
+            statement.setLong(1, order.getId());
+            statement.executeUpdate();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't update order "
+                    + order, e);
+        }
+        addProductsToOrder(order);
+        return order;
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String query = "UPDATE orders SET deleted = true WHERE order_id = ?";
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(query);
+            statement.setLong(1, id);
+            return statement.executeUpdate() == 1;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't delete order with Id = " + id, e);
+        }
+    }
+
+    private Order addProductsToOrder(Order order) {
+        String query = "INSERT INTO order_products(order_id, user_id) VALUES (?, ?);";
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(query);
+            statement.setLong(1, order.getId());
+            for (Product product : order.getProducts()) {
+                statement.setLong(2, product.getId());
+                statement.executeUpdate();
+            }
+            return order;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Failed to add products to order with Id = "
+                    + order.getId(), e);
+        }
+    }
+
+    private Order getOrderFromResultSet(ResultSet resultSet) throws SQLException {
+        long orderId = resultSet.getLong("order_id");
+        long userId = resultSet.getLong("user_id");
+        return new Order(orderId, userId);
+    }
+
+    private List<Product> getProductsFromOrder(Long orderId) {
+        String query = "SELECT * FROM products JOIN order_products  "
+                + "ON order_products.product_id = products.product_id "
+                + "WHERE order_products.order_id = ?;";
```

Comment:
You could ALIAS where you mention several tables in one query, e.g.
```suggestion
        String query = "SELECT * FROM products p JOIN order_products op  "
                + "ON op.product_id = p.product_id "
                + "WHERE op.order_id = ?;";
```